src/votekit/models.py

```diff
@@ -1,95 +1,292 @@
+        """
+        Fetch the PreferenceProfile of the given round number.
+
+        Args:
+            round_number (int, optional): The round number. Defaults to -1,
+                which accesses the final profile.
+
+        Returns:
+            PreferenceProfile
+
+        """
+        if round_number < -1:
+            raise ValueError("round_number must be -1 or non-negative.")
+
+        if round_number == -1:
+            round_number = len(self)
+
+        profile = self._profile
+
+        for i in range(round_number):
+            profile = self._run_step(profile, self.election_states[i])
+
+        return profile
+
+    def get_step(
+        self, round_number: int = -1
+    ) -> tuple[PreferenceProfile, ElectionState]:
+        """
+        Fetches the profile and ElectionState of the given round number.
+
+        Args:
+            round_number (int, optional): The round number to fetch. Defaults to -1,
+                which fetches the final round.
+
+        Returns:
+            tuple[PreferenceProfile, ElectionState]
+        """
+        if round_number < -1:
+            raise ValueError("round_number must be -1 or non-negative.")
+
+        if round_number == -1:
+            round_number = len(self)
+
+        return (self.get_profile(round_number), self.election_states[round_number])
+
+    def get_elected(self, round_number: int = -1) -> list[frozenset[str]]:
+        """
+        Fetch the elected candidates up to the given round number.
+
+        Args:
+            round_number (int, optional): Round number, defaults to -1 which corresponds to
+                the final round.
+
+        Returns:
+            list[frozenset[str]]: List of winning candidates in order of election. Candidates
+                in the same set were elected simultaneously, i.e. in the final ranking
+                they are tied.
+        """
+        if round_number < -1:
+            raise ValueError("round_number must be -1 or non-negative.")
+
+        if round_number == -1:
+            round_number = len(self)
+
+        return [
+            s
+            for state in self.election_states[: (round_number + 1)]
+            for s in state.elected
+            if state.elected != (frozenset(),)
+        ]
+
+    def get_eliminated(self, round_number: int = -1) -> list[frozenset[str]]:
+        """
+        Fetch the eliminated candidates up to the given round number.
+
+        Args:
+            round_number (int, optional): Round number, defaults to -1 which corresponds to
+                the final round.
 
-        self.state = ElectionState(curr_round=0, profile=self._profile)
+        Returns:
+            list[frozenset[str]]: List of eliminated candidates in reverse order of elimination.
+                Candidates in the same set were eliminated simultaneously, i.e. in the final ranking
+                they are tied.
+        """
+        if round_number < -1:
```

Comment:
Again, let the list handle the error here

Reply:
If you want to make your own error message, you can catch the IndexError that the list raises, and then raise a new index error with a custom message from that